osbrain/agent.py

```diff
@@ -1697,10 +1697,15 @@ def run(self):
             self.queue.put(format_exception())
             return
 
+        uri = self._daemon.register(self.agent)
+        try:
+            ns.register(self.name, uri, safe=True)
+        except Pyro4.errors.NamingError as error:
+            self.queue.put(str(error))
+            return
```

Comment:
Looks good. Two comments though:

- Maybe instead of `str(error)` you could use `format_exception()`?
- You could add a `finally: ns.release()` clause (and remove the current `ns.release()` call bellow).